lib/ts/recipe/multifactorauth/index.ts

```diff
@@ -0,0 +1,166 @@
+/* Copyright (c) 2024, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import Recipe from "./recipe";
+import { RecipeInterface, APIOptions, APIInterface } from "./types";
+import { MultiFactorAuthClaim } from "./multiFactorAuthClaim";
+import { SessionContainerInterface } from "../session/types";
+import { getUser } from "../..";
+import { getUserContext } from "../../utils";
+import { getMFARelatedInfoFromSession } from "./utils";
+
+export default class Wrapper {
+    static init = Recipe.init;
+
+    static MultiFactorAuthClaim = MultiFactorAuthClaim;
+
+    static async assertAllowedToSetupFactorElseThrowInvalidClaimError(
+        session: SessionContainerInterface,
+        factorId: string,
+        userContext?: Record<string, any>
+    ) {
+        let ctx = getUserContext(userContext);
+
+        const mfaInfo = await getMFARelatedInfoFromSession({
+            session,
+            assumeEmptyCompletedIfNotFound: false,
+            userContext: ctx,
+        });
+
+        if (mfaInfo.status === "SESSION_USER_NOT_FOUND_ERROR") {
+            throw new Error("Session user not found");
+        } else if (mfaInfo.status === "MFA_CLAIM_VALUE_NOT_FOUND_ERROR") {
+            throw new Error("MFA claim value not found");
```

Comment:
ideally the session should be updated here to have the mfa claim and then we run the logic with that!